Error while checking build/O2/fullCI for da56437657f1d0552cfecc4a9c6b3614f5568ade at 2023-07-04 16:13:
```
## sw/BUILD/GEANT4_VMC-latest/log
CMake Error at /sw/slc8_x86-64/CMake/v3.23.1-13/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11609/da56437657f1d0552cfecc4a9c6b3614f5568ade/build_O2_fullCI/pretty.html).

--- a/Detectors/TRD/workflow/src/TrackBasedCalibSpec.cxx
+++ b/Detectors/TRD/workflow/src/TrackBasedCalibSpec.cxx
@@ -141,9 +141,9 @@ DataProcessorSpec getTRDTrackBasedCalibSpec(o2::dataformats::GlobalTrackID::mask
   if (GTrackID::includesSource(GTrackID::Source::TPC, src)) {
     LOGF(info, "Found TPC tracks as input, loading TPC-TRD");
     srcTrk |= GTrackID::getSourcesMask("TPC-TRD");
-    if (gain) {
-      srcTrk |= GTrackID::getSourcesMask("TPC");
-    }
+  }
+  if (gain) {
+    srcTrk |= GTrackID::getSourcesMask("TPC");
   }
   GTrackID::mask_t srcClu = GTrackID::getSourcesMask("TRD");         // we don't need all clusters, only TRD tracklets
   dataRequest->requestTracks(srcTrk, false);

--- a/prodtests/full-system-test/aggregator-workflow.sh
+++ b/prodtests/full-system-test/aggregator-workflow.sh
@@ -64,6 +64,7 @@ if [[ "${GEN_TOPO_VERBOSE:-}" == "1" ]]; then
   echo "CALIB_PHS_RUNBYRUNCALIB = $CALIB_PHS_RUNBYRUNCALIB" 1>&2
   echo "CALIB_PHS_L1PHASE = $CALIB_PHS_L1PHASE" 1>&2
   echo "CALIB_TRD_VDRIFTEXB = $CALIB_TRD_VDRIFTEXB" 1>&2
+  echo "CALIB_TRD_GAIN = $CALIB_TRD_GAIN" 1>&2
   echo "CALIB_TPC_TIMEGAIN = $CALIB_TPC_TIMEGAIN" 1>&2
   echo "CALIB_TPC_RESPADGAIN = $CALIB_TPC_RESPADGAIN" 1>&2
   echo "CALIB_TPC_SCDCALIB = $CALIB_TPC_SCDCALIB" 1>&2
@@ -211,8 +212,15 @@ if [[ $AGGREGATOR_TASKS == BARREL_TF ]] || [[ $AGGREGATOR_TASKS == ALL ]]; then
     add_W o2-tpc-vdrift-tgl-calibration-workflow ""
   fi
   # TRD
+  TRD_CALIB_CONFIG=
   if [[ $CALIB_TRD_VDRIFTEXB == 1 ]]; then
-    add_W o2-calibration-trd-workflow "--vDriftAndExB"
+    TRD_CALIB_CONFIG+=" --vDriftAndExB"
+  fi
+  if [[ $CALIB_TRD_GAIN == 1 ]]; then
+    TRD_CALIB_CONFIG+=" --gain"
+  fi
+  if [[ ! -z ${TRD_CALIB_CONFIG} ]]; then
+    add_W o2-calibration-trd-workflow "${TRD_CALIB_CONFIG}"
   fi
 fi
 

--- a/prodtests/full-system-test/dpl-workflow.sh
+++ b/prodtests/full-system-test/dpl-workflow.sh
@@ -158,6 +158,7 @@ fi
 
 
 workflow_has_parameter CALIB && [[ $CALIB_TRD_VDRIFTEXB == 1 ]] && TRD_CONFIG+=" --enable-vdexb-calib"
+workflow_has_parameter CALIB && [[ $CALIB_TRD_GAIN == 1 ]] && TRD_CONFIG+=" --enable-gain-calib"
 ! has_detector FT0 && TRD_CONFIG+=" --disable-ft0-pileup-tagging"
 
 SEND_ITSTPC_DTGL=

